### PR TITLE
Fix sanitizer out-of-memory error

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Change max array element check in cnpy to avoid out-of-memory error in sanitizer tests

### Tickets:
 - CVS-105669

### CODE REVIEW

File: thirdparty/cnpy/cnpy.h [hunk 1]
```diff
@@ -18,17 +18,36 @@
...
+        }
+
         NpyArray(const std::vector<size_t>& _shape, size_t _word_size, bool _fortran_order) :
             shape(_shape), word_size(_word_size), fortran_order(_fortran_order)
         {
             num_vals = 1;
             for(size_t i = 0;i < shape.size();i++) num_vals *= shape[i];
             if (word_size &&
-                num_vals > std::vector<char>().max_size() / word_size)
+                num_vals > (0.9 * GetFreeMemorySize() / sizeof(char) / word_size))
```
Comment (ilyachur) [inline thirdparty/cnpy/cnpy.h:50]: Why do we need these constants?
What was the original issue?

**Comment by ilyachur** [comment]
Please fix CI

**Comment by ilyachur** [comment]
Please fix CI

File: thirdparty/cnpy/cnpy.h [hunk 2]
```diff
@@ -1,294 +1,353 @@
-//Copyright (C) 2011  Carl Rogers
-//Released under MIT License
-//license available in LICENSE file, or at http://www.opensource.org/licenses/mit-license.php
+// Copyright (C) 2011  Carl Rogers
```
Comment (ilyachur) [inline thirdparty/cnpy/cnpy.h:1]: Could you please revert code style changes?

Comment (olpipi) [reply thirdparty/cnpy/cnpy.h]: done

**Review by ilyachur — APPROVED** [review]

File: thirdparty/cnpy/cnpy.h [hunk 3]
```diff
@@ -19,21 +19,49 @@
...
-                new std::vector<char>(num_vals * word_size));
+            try {
+                data_holder = std::shared_ptr<std::vector<char>>(
+                    new std::vector<char>(num_vals * word_size));
+            } catch (std::bad_alloc const &) {
+                throw std::length_error("NpyArray of " + std::to_string(num_vals) +
+                                        "*" + std::to_string(word_size) +
+                                        " elements is too big.");
+            }
         }
```
Comment (ilya-lavrenov) [inline thirdparty/cnpy/cnpy.h:65]: better to move implementation of ctor in cpp file to aviud system headers in .h

**Pull request merged**